addons/l10n_fr_pos_cert/models/pos.py

```diff
@@ -0,0 +1,206 @@
+from datetime import datetime
+from hashlib import sha256
+from json import dumps
+import pytz
+
+from openerp import models, api, fields
+from openerp.fields import Datetime
+from openerp.tools.translate import _
+from openerp.exceptions import UserError
+
+NOT_SAME_DAY_ERROR = _("This session has been opened another day. To comply with the French law, you should close sessions on a daily basis. Please close session %s and open a new one.")
+
+
+def ctx_tz(record, field):
+    res_lang = None
+    ctx = record._context
+    tz_name = pytz.timezone(ctx.get('tz') or record.env.user.tz)
+    timestamp = Datetime.from_string(record[field])
+    if ctx.get('lang'):
+        res_lang = record.env['res.lang'].search([('code', '=', ctx['lang'])], limit=1)
+    if res_lang:
+        timestamp = pytz.utc.localize(timestamp, is_dst=False)
+        return datetime.strftime(timestamp.astimezone(tz_name), res_lang.date_format + ' ' + res_lang.time_format)
+    return Datetime.context_timestamp(record, timestamp)
+
+
+class pos_config(models.Model):
+    _inherit = 'pos.config'
+
+    @api.multi
+    def open_ui(self):
+        date_today = datetime.utcnow()
+        for config in self.filtered(lambda c: c.company_id._is_accounting_unalterable()):
+            if config.current_session_id:
+                session_start = Datetime.from_string(config.current_session_id.start_at)
+                if session_start.date() != date_today.date():
+                    raise UserError(NOT_SAME_DAY_ERROR % config.current_session_id.name)
```

Comment:
I agree with the 24h threshold, but in any case this is a constraint (and intended to be). So changes in workflows should occur anyway

Also, we only do a "soft" constraint. only re-opening a session is forbidden. but keeping an old one open is still allowed.

Reply:
> only re-opening a session is forbidden

I think that's a problem for bar / restaurant. If user loose his session during the night, (Browser trouble, computer reboot), he will not have the possibility to reuse it, and you'll force users to close / open a session during his shift, that's not possible.

> but keeping an old one open is still allowed.

for how long ? Because if he's keeping a session opened during one week, the module does'nt feat with the law I think.

What about the proposal (based on OCA/pos#34) ?
(If it's just a matter of time, I can contribute.)

Reply:
It seems to me that blocking new pos orders might be annoying too for users facing a waiting line or a rush. I would therefore only block the start of a new session IF this session is opened from more than 24 hours. So we keep our check mechanism but with your condition.